### ISSUE TITLE
We should really think twice about uint64 vs int64 usage for timestamp storage

### ISSUE THREAD

**Issue by MartinBriza** [toggl/toggldesktop#3005]
While fixing warnings in the library, I noticed the pattern of storing almost every time value as `Poco::UInt64` (unsigned), except for the `Duration` field in some structures, that is `Poco::Int64` (signed). There is a pattern of implicit casting of those values in various calls, most notably for `       time_t time(time_t *tloc);` - `time_t` is a signed value and I think we should stick to that.

Using unsigned values when subtracting and comparing them is bug-prone and inconsistent with a lot of system functions. I suggest we port fields like `StartTime` (and many others) to `Poco::Int64`

**Comment by IndrekV** [comment]
Duration is signed int because it can be negative. It is negative when the timer is running. If there are no clear benefits ( I can't think of any right now ) to keeping everything as unsigned int's then I think moving to `signed int` is a good idea.

**Comment by MartinBriza** [comment]
Yeah, from my point of view, usage of signed integers has clear benefits, not just because of running entries but overall for working with time differences and being able to compare the values without casting.

**Comment by IndrekV** [comment]
I agree that casting should be avoided if possible. Good catch @MartinBriza. The library overhaul is producing great improvements already.
